### ISSUE TITLE
Is it possible to change the amount of conflict allowed for Exclusive Feature Bundling?

### ISSUE THREAD

**Issue by jameslamb** [microsoft/LightGBM#3591]
[The original LightGBM paper](https://papers.nips.cc/paper/2017/file/6449f44a102fde848669bdd9eb6b76fa-Paper.pdf) mentions using an approximation of mutual exclusivity in Exclusive Feature Bundling (EFB), like this:

>  we notice that there are usually quite a few features, although not 100% mutually exclusive, also rarely
take nonzero values simultaneously. If our algorithm can allow a small fraction of conflicts, we can
get an even smaller number of feature bundles and further improve the computational efficiency.

This made me think that the amount of allowed conflict could be something that is tuned, since there seems to be a trade-off: higher allowed amount of conflict should mean faster training, but possibly at the expense of some information loss.

I don't see this documented as a parameter in https://lightgbm.readthedocs.io/en/latest/Parameters.html, and it looks like it is hard-coded to `#data / 10000` in https://github.com/microsoft/LightGBM/blob/0655d67cc1ee28d536a773a6c5d978d38a0e56c1/src/io/dataset.cpp#L106-L108.

1. Is it possible to change this level of conflict directly?
1. Should LightGBM support configuring this via parameters? Or is there something I'm missing that makes it a bad idea to expose this parameter?

**Comment by StrikerRUS** [comment]
For `#1` answer is
> Currently we don't provide $K$ or $\gamma$ as a hyper-parameter. Instead, $K$ is hard-coded as `num_data / 10000` in the code.
   https://github.com/microsoft/LightGBM/issues/4114#issuecomment-813201652

**Comment by StrikerRUS** [comment]
@jameslamb Would you like to turn this into a feature request?

**Comment by jameslamb** [comment]
Only if you or @shiyu1994 think it's a good idea to try tuning this parameter. I'm hesitant to add more parameters to LightGBM unless there is a compelling reason to do so.

**Comment by shiyu1994** [comment]
Now we have `MultiValBin` class which allows a feature group to contain multiple values in the same row. With too many conflicts in a group, a better solution is to transform it into a multi-value group. A sparse multi-value group also exploits sparsity and accelerates training.
So, although we hard code the maximum allowed conflicts, those features with no proper single-value feature group can end up in the sparse multi-value feature group.
Overall, I think currently we don't need to expose this parameter.

**Comment by jameslamb** [comment]
Thanks as always for the detailed explanation @shiyu1994 ! I'll close this then and I don't think we need to add it as a feature request.

**Comment by StrikerRUS** [comment]
Just for the history: `max_conflict_rate` param was removed in #2699.
https://github.com/microsoft/LightGBM/commit/509c2e50c25eded99fc0997afe25ebee1b33285d#diff-229636b798e2650929dad136deec0c435e57b33b41c48d08eb32b2661e2c5e84L537-L542
